guides/source/active_model_basics.md

```diff
@@ -122,264 +457,766 @@ class Person
+log_me method: called around the update method
+update method called
+log_me method: block successfully called
+finalize_me method: called after the update method
+=> nil
+```
+
+As per the above example, when defining an 'around' callback remember to `yield`
+to the block, otherwise, it won't be executed.
+
+NOTE: `method_name` passed to `define_model_callbacks` must not end with `!`,
+`?` or `=`. In addition, defining the same callback multiple times will
+overwrite previous callback definitions.
+
+#### Defining Specific Callbacks
+
+You can choose to create specific callbacks by passing the `only` option to the
+`define_model_callbacks` method:
+
+```ruby
+define_model_callbacks :update, :create, only: [:after, :before]
+```
+
+This will create only the `before_create` / `after_create` and `before_update` /
+ `after_update` callbacks, but skip the `around_*` ones. The option will apply
+to all callbacks defined on that method call. It's possible to call
+`define_model_callbacks` multiple times, to specify different lifecycle events:
+
+```ruby
+define_model_callbacks :create, only: :after
+define_model_callbacks :update, only: :before
+define_model_callbacks :destroy, only: :around
+```
+
+This will create `after_create`, `before_update`, and `around_destroy` methods
+only.
+
+#### Defining Callbacks with a Class
+
+You can pass a class to `before_<type>`, `after_<type>` and `around_<type>` for
+more control over when and in what context your callbacks are triggered. The
+callback will trigger that class's `<action>_<type>` method, passing an instance
+of the class as an argument.
+
+```ruby
+class Person
+  extend ActiveModel::Callbacks
+
+  define_model_callbacks :create
+  before_create PersonCallbacks
+end
+
+class PersonCallbacks
+  def self.before_create(obj)
+    # `obj` is the Person instance that the callback is being called on
   end
 end
 ```
 
+#### Aborting Callbacks
+
+The callback chain can be aborted at any point in time by throwing `:abort`.
+This is similar to how Active Record callbacks work.
+
+In the example below, since we throw `:abort` before an update in the `reset_me`
+method, the remaining callback chain including `before_update` will be aborted,
+and the body of the `update` method won't be executed.
+
+```ruby
+class Person
+  extend ActiveModel::Callbacks
+
+  define_model_callbacks :update
+
+  before_update :reset_me
+  after_update :finalize_me
+  around_update :log_me
+
+  def update
+    run_callbacks(:update) do
+      puts "update method called"
+    end
+  end
+
+  private
+    def reset_me
+      puts "reset_me method: called before the update method"
+      throw :abort
+      puts "reset_me method: some code after abort"
+    end
+
+    def finalize_me
+      puts "finalize_me method: called after the update method"
+    end
+
+    def log_me
+      puts "log_me method: called around the update method"
+      yield
+      puts "log_me method: block successfully called"
+    end
+end
+```
+
+```
```

Comment:
<details><summary>For proper highlighting:</summary>
<p>

![Screenshot 2024-03-06 at 17 20 13](https://github.com/rails/rails/assets/26328/17207f7d-0252-4721-a2c5-864a8a69429d)

</p>
</details> 

````suggestion
```irb
````